include/riak_kv_sweeper.hrl

```diff
@@ -0,0 +1,65 @@
+%% Copyright (c) 2011-2015 Basho Technologies, Inc.  All Rights Reserved.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+
+%% Used as configuration for sweep participants.
+-record(sweep_participant,
```

Comment:
So a couple weeks ago, I already tried to put it into its own module...but my first attempt really made the code a lot uglier and less readable, because then we can no longer pattern match on record fields in function heads. In order to fix this is a nice clean way, we're going to have to rewrite a pretty substantial amount of code.

If I had written this code, I would not have used shared records all over the place, but I am questioning whether it is worth the effort and trouble to rewrite the sweeper at this point, given that it works and we are under a deadline crunch. I can do it, but it will not be a trivial change.

Also consider that these are not records like e.g. riak_object, where they are passed around all over the system. All these shared records are mostly just used by a few tightly-coupled sweeper modules. They are also not passed across node boundaries via RPC, so there should be no risk of upgrade problems involving mixed clusters with differing record definitions.